tests/python_package_test/test_basic.py

```diff
@@ -678,10 +678,22 @@ def test_list_to_1d_numpy(collection, dtype, rng):
     custom_name = "my_custom_variable"
 
     if collection.startswith("pd"):
-        if not PANDAS_INSTALLED:
-            pytest.skip("pandas is not installed")
-        else:
-            y = pd_Series(y)
+        pd = pytest.importorskip("pandas")
+        y = pd_Series(y)
+        if pd.api.types.is_object_dtype(y):
+            with pytest.raises(
+                ValueError,
+                match=r"pandas dtypes must be int, float or bool\.\nFields with bad pandas dtypes: 0: object",
+            ):
+                lgb.basic._list_to_1d_numpy(y, dtype=np.float32, name=custom_name)
+            return
+        elif pd.api.types.is_string_dtype(y):
```

Comment:
This change fixes another CI-blocking issue... described in https://github.com/microsoft/LightGBM/issues/6980#issuecomment-3121313890

I'm proposing switching to `pd = pytest.importorskip("pandas")` here so the `pandas` dtype-checking functions can be used in the test. Those have been in `pandas` since `pandas==1.0.0`, and the oldest version `lightgbm` is tested against is `1.1.3`

https://github.com/microsoft/LightGBM/blob/a0fde1b00d40eb664945aeaca8c9159ff872206b/.ci/test-python-oldest.sh#L16

so this should be safe